src/world/generator/object/Tree.php

```diff
@@ -100,11 +102,22 @@ public function canPlaceObject(ChunkManager $world, int $x, int $y, int $z, Rand
 		return true;
 	}
 
-	public function placeObject(ChunkManager $world, int $x, int $y, int $z, Random $random) : void{
+	public function placeObject(ChunkManager $world, int $x, int $y, int $z, Random $random, bool $callEvent = false) : void{
 		$transaction = new BlockTransaction($world);
 		$this->placeTrunk($x, $y, $z, $random, $this->generateChunkHeight($random), $transaction);
 		$this->placeCanopy($x, $y, $z, $random, $transaction);
 
+		if($callEvent){
+			$ev = new BlockSproutEvent(
+				$world->getBlockAt($x, $y, $z),
+				$transaction
+			);
```

Comment:
Seems unnecessary to spread this over multiple lines now, no?

Reply:
It felt a little bit crowded for me if everything was in one line, but I will change it if you are ok with it.